x-pack/plugins/task_manager/server/lib/create_managed_configuration.ts

```diff
@@ -95,14 +100,41 @@ function createPollIntervalScan(logger: Logger, startingPollInterval: number) {
     if (errorCount > 0) {
       // Increase poll interval by POLL_INTERVAL_INCREASE_PERCENTAGE and use Math.ceil to
       // make sure the number is different than previous while not being a decimal value.
-      newPollInterval = Math.ceil(previousPollInterval * POLL_INTERVAL_INCREASE_PERCENTAGE);
+      // Also ensure we don't go over PREFERRED_MAX_POLL_INTERVAL or startingPollInterval,
+      // whichever is greater.
+      newPollInterval = Math.min(
+        Math.ceil(previousPollInterval * POLL_INTERVAL_INCREASE_PERCENTAGE),
+        Math.max(PREFERRED_MAX_POLL_INTERVAL, startingPollInterval)
+      );
+      if (
```

Comment:
If we coorce these values to integers (via round/ceil/floor) we can use [`Number.isSafeInteger()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger`) which covers the cases below, and more (< `MIN_SAFE_INTEGER`).

Reply:
@pmuellr I added the extra safety check here: https://github.com/elastic/kibana/pull/172394/commits/121e942274704a483eac98a3c56abeaa0b6923e2

It should cover the lower end of things as well, on top of the `< 0` check.

Regarding ensuring they are integers, I think we have the grounds covered after this commit: https://github.com/elastic/kibana/pull/172394/commits/61364003e5200c263f6f19c4fecb3b41f744726e

Reply:
`isSafeInteger()` also covers: NaN, typeof != 'number', so I think we can get rid of those checks as well!  Seems like all we need is that and a negative check.

Reply:
Nice! TIL, made the further optimization here: https://github.com/elastic/kibana/pull/172394/commits/e0ddce3ccac940aac1f2e55cf38d6f916ed125e3 and switched the order of checks for readability here https://github.com/elastic/kibana/pull/172394/commits/816a85df9aa9a74c13b5ed40a79fbe0e2401dada